### ISSUE TITLE
data augmentation issues on summarization

### ISSUE THREAD

**Issue by monologue1107** [pytorch/fairseq#2886]
Hi, sorry to bother.
I am supposed to manually add some other tags after sentences in each line in train.target for loss computation during training. 
`For example, the original line(summary) in train.target file is "hello, world." //The fixed line is supposed to be "hello, world. <token> positive."` 
(I am about to insert some special tokens during them.However, I can't let the added part influence the training or fine-tuning. 
As my initial idea, I want to add a value-key in dict in LanguagePairDataset so that in loss part I can use sample['target_2'] to find my adding tags such as 'positive'.(sample['target'] is the reference summary)
So how can I fix this issue? Fairseq framework is packaged so well that I can't handle it roughly. I would like to hear for some tips or advice on this idea. Thanks a lot!

**Comment by monologue1107** [comment]
@myleott @alexeib @tangyuq

**Comment by huihuifan** [comment]
zero the loss in the NLL criterion for the special token :) just like it zeroes the padding token

**Comment by monologue1107** [comment]
> zero the loss in the NLL criterion for the special token :) just like it zeroes the padding token

sorry to bother, what 'special token' do you think is the most suitable for the training which will not cause misunderstandings
.
